src/test/java/core/basesyntax/CalculatorTest.java

```diff
@@ -0,0 +1,234 @@
+
+    @Test
+    void calculate_additionMinMaxDouble_ok() {
+        double expected = Double.MAX_VALUE + Double.MIN_VALUE;
+        double actual = calculator.calculate(Double.MAX_VALUE, Double.MIN_VALUE, ADDITION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_subtractionTwoPositive_ok() {
+        double expected = 1;
+        double actual = calculator.calculate(5, 4, SUBTRACTION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_subtractionTwoNegative_ok() {
+        double expected = -3;
+        double actual = calculator.calculate(-5, -2, SUBTRACTION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_subtractionPositiveNegative_ok() {
+        double expected = 30;
+        double actual = calculator.calculate(10, -20, SUBTRACTION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_subtractionFirstZero_ok() {
+        double expected = -5;
+        double actual = calculator.calculate(0, 5, SUBTRACTION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_subtractionSecondZero_ok() {
+        double expected = 4;
+        double actual = calculator.calculate(4, 0, SUBTRACTION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_subtractionMinMaxDouble_ok() {
+        double expected = Double.MAX_VALUE - Double.MIN_VALUE;
+        double actual = calculator.calculate(Double.MAX_VALUE,
+                                             Double.MIN_VALUE,
+                                             SUBTRACTION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_multiplicationTwoPositive_ok() {
+        double expected = 2;
+        double actual = calculator.calculate(1, 2, MULTIPLICATION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_multiplicationTwoNegative_ok() {
+        double expected = 10;
+        double actual = calculator.calculate(-5, -2, MULTIPLICATION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_multiplicationPositiveNegative_ok() {
+        double expected = -40;
+        double actual = calculator.calculate(2, -20, MULTIPLICATION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_multiplicationFirstZero_ok() {
+        double expected = 0;
+        double actual = calculator.calculate(0, 5, MULTIPLICATION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_multiplicationSecondZero_ok() {
+        double expected = 0;
+        double actual = calculator.calculate(4, 0, MULTIPLICATION);
+        assertEquals(expected, actual, DELTA);
+    }
+
+    @Test
+    void calculate_multiplicationMinMaxDouble_ok() {
+        double sum = calculator.calculate(Double.MAX_VALUE, Double.MIN_VALUE, MULTIPLICATION);
```

Comment:
```suggestion
        double product/actual = calculator.calculate(Double.MAX_VALUE, Double.MIN_VALUE, MULTIPLICATION);
```
https://en.wikipedia.org/wiki/Product_(mathematics)